src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props

```diff
@@ -45,15 +45,24 @@ Copyright (c) .NET Foundation. All rights reserved.
                       Publish="true" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
-    <PackageReference Include="Microsoft.NETCore.App" Version="$(RuntimeFrameworkVersion)" IsImplicitlyDefined="true" />
+  <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(RuntimeFrameworkName)' != ''">
+    <PackageReference Include="$(RuntimeFrameworkName)">
+      <Version>$(RuntimeFrameworkVersion)</Version>
+      <IsImplicitlyDefined>true</IsImplicitlyDefined>
+      <Publish>true</Publish>
+      <!-- 
+        By default, do not include the runtime framework package in the generated .nuspec when packing the project.
+        The exception is for DotNetCliTool projects, which need the runtime package listed as a dependency.
+      -->
+      <PrivateAssets Condition=" '$(RuntimeFrameworkName)' != 'Microsoft.NETCore.App' AND '$(PackageType)' != 'DotNetCliTool' ">All</PrivateAssets>
+    </PackageReference>
 
     <!-- For libraries targeting .NET Core 2.0 or higher, don't include a dependency on Microsoft.NETCore.App in the package produced by pack.
          Packing an app (for example a .NET CLI tool) should include the Microsoft.NETCore.App package dependency. -->
     <PackageReference Update="Microsoft.NETCore.App"
                       Condition="('$(OutputType)' != 'Exe') And ('$(_TargetFrameworkVersionWithoutV)' != '') And ('$(_TargetFrameworkVersionWithoutV)' >= '2.0')"
                       PrivateAssets="All"
-                      Publish="true" />    
+                      Publish="true" />  
```

Comment:
The Publish=true is redundant now, right? 

It feels like this should also be a conditional entry on the PackageReference. I'm having trouble following the rationale for Update in this case, but not for RuntimeFrameworkName != Microsoft.NETCore.App.

I suppose there could be a behavior change if user has explicit package ref and DisableImplict != true.

Reply:
The reason for this was that PrivateAssets="All" on NETCore.App only if OutputType == library, but for AspNetCore.App, PrivateAssets is always All (when using the implicit package version.)

I think the resolution to #2364 is to actually tweak this line so PrivateAsset="All" is always true unless PackageType== DotNetCliTool, but that's a separate change to make...didn't want to muddy the PR.

Reply:
https://github.com/dotnet/sdk/pull/2395 is out that I think would fix #2364. Since the pain is similar, I think there's a case to be made that we should pull in #2395 into the same release as this. It would end up making the code easier to follow. Something for us to consider, but not feedback for your change. cc @wli3

Reply:
Cool, glad that's going in to 2.2. 

Also, yes, the Publish="true" is redundant. I can remove that line.